src/core/Akka.TestKit/Internal/BufferQueue.cs

```diff
@@ -105,19 +109,22 @@ public bool TryTake(out T item, int millisecondsTimeout, CancellationToken cance
         /// This exception is thrown when the operation is canceled.
         /// </exception>
         /// <returns>The item removed from the collection.</returns>
-        public T Take(CancellationToken cancellationToken)
+        public async ValueTask<T> Take(CancellationToken cancellationToken)
         {
-            var p = _collection.Take(cancellationToken);
+            var p = await _collection.ReceiveAsync(cancellationToken);
             return p.Value;
         }
 
         /// <summary>
-        /// Copies the items from the <see cref="BlockingQueue{T}"/> instance into a new <see cref="List{T}"/>.
+        /// Copies the items from the <see cref="BufferQueue{T}"/> instance into a new <see cref="List{T}"/>.
         /// </summary>
         /// <returns>A <see cref="List{T}"/> containing copies of the elements of the collection</returns>
         public List<T> ToList()
         {
-            var positionArray = _collection.ToArray();
+            //IAsyncEnumerable<T> NOT SUPPORTED IN .NET Standards 2.1
+
+            // https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.dataflow.bufferblock-1.tryreceiveall?view=net-6.0#applies-to
+            _collection.TryReceiveAll(out var positionArray);
```

Comment:
we can read a collection of messages

Reply:
This will fail. `TryReceiveAll` will empty the queue.